doc/man3/SSL_CTX_set_cert_verify_callback.pod

```diff
@@ -31,13 +31,20 @@ When the verification must be performed, I<callback> will be called with
 the arguments callback(X509_STORE_CTX *x509_store_ctx, void *arg). The
 argument I<arg> is specified by the application when setting I<callback>.
 
-I<callback> should return 1 to indicate verification success and 0 to
-indicate verification failure. If SSL_VERIFY_PEER is set and I<callback>
-returns 0, the handshake will fail. As the verification procedure may
-allow the connection to continue in the case of failure (by always
-returning 1) the verification result must be set in any case using the
-B<error> member of I<x509_store_ctx> so that the calling application
-will be informed about the detailed result of the verification procedure!
+I<callback> should return 1 to indicate verification success
+and 0 to indicate verification failure.
+If B<SSL_VERIFY_PEER> is set and I<callback> returns 0, the handshake will fail.
+The I<callback> may also return -1 to indicate that the handshake
+shall suspend and return control to the calling application
+with B<SSL_ERROR_WANT_RETRY_VERIFY>.
+The app can for instance fetch further certificates or cert status information
+needed for the verification and then call L<SSL_connect(3)>) again.
+This resumes the connection attempt by retrying the cert verification step.
```

Comment:
```suggestion
This resumes the connection attempt by retrying the failed cert verification step.
```

or perhaps

```suggestion
This resumes the connection attempt by retrying the previously failed cert verification step.
```

Rationale: Referring to the verification step as failed may help folks realize that while the handshake as a whole has been suspended, its verification step was _not_ suspended. The verification step is considered failed in this case. That, in turn, explains why the callback has to set an error even when returning `-1` and suspending negotiation (as a whole) with a "soft" SSL_ERROR_WANT_RETRY_VERIFY.

Reply:
At this point the formulation was correct because the retry could be done also in case there was no previously failed verification. Instead, I've made clear a little further up that the `-1` is usually only returned on failure verifying the server certificate.

Reply:
> the retry could be done also in case there was no previously failed verification

Good point! I wish I could suggest another way to emphasize that the verification step, unlike the handshake, has not been suspended and will be retried _from scratch_. I do not have any good suggestions, and the existing PR changes are not wrong here, so I consider this change request addressed. Unfortunately, GitHub does not let me mark this change request as resolved for some reason.

Reply:
> I wish I could suggest another way to emphasize that the verification step, unlike the handshake, has not been suspended and will be retried _from scratch_. I do not have any good suggestions,

I'd say this point is already sufficiently addressed by the following sentence:
```
Calling L<SSL_connect(3)> again resumes the connection attempt
by retrying the server certificate verification step.
```
(though it might have been stressed by inserting "_from scratch_").


> and the existing PR changes are not wrong here, so I consider this change request addressed. Unfortunately, GitHub does not let me mark this change request as resolved for some reason.

Anyway your change request has disappeared,
and Matt's approval is there, so no problem.